### PR TITLE
Spec for Geofence API

### PR DESCRIPTION
**Pull request by BindyaBhaskar**
Added spec for : 

- List all geofences.

- Get information of a geofence.

- Delete a geofence.

### CODE REVIEW

File: spec/openapi.yaml [hunk 1]
```diff
@@ -98,6 +98,8 @@ tags:
     externalDocs:
       description: Find out more about Compliance Policy
       url: 'https://docs.esper.io/'
+  - name: Geofence
+    description: APIs for geofence management
```
Comment (tr0n-p46) [inline spec/openapi.yaml:102]: Please add external docs
https://docs.esper.io/home/console.html#geofence

File: spec/paths/v0@enterprise@{enterprise_id}@geofence@{geofence_id}@.yaml [hunk 2]
```diff
@@ -0,0 +1,44 @@
+get:
```
Comment (tr0n-p46) [inline spec/paths/v0@enterprise@{enterprise_id}@geofence@{geofence_id}@.yaml:1]: Add query params for GET requests

Comment (BindyaBhaskar) [reply spec/paths/v0@enterprise@{enterprise_id}@geofence@{geofence_id}@.yaml:1]: Will do.

File: spec/openapi.yaml (hunk 1 above)
Comment (BindyaBhaskar) [reply spec/openapi.yaml:102]: Okay

File: spec/definitions/Geofence.yaml [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+    enum:
+      - KILOMETERS
+      - METERS
+      - MILES
+  device_actions:
+    title: Device Actions
+    description: The actions to be performed on the device
+    type: array
+    items:
+      type: string
```
Comment (tr0n-p46) [inline spec/definitions/Geofence.yaml:48]: Please add the enums also for device_actions ie. LOCK and BEEP

File: spec/paths/v0@enterprise@{enterprise_id}@geofence@.yaml [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+            type: array
+            items:
+              $ref: '#/definitions/Geofence'
+  security:
+    - apiKey: []
+  tags:
+    - Geofence
+
+post:
+  operationId: uploadGeofence
```
Comment (abdulMuneerM) [inline spec/paths/v0@enterprise@{enterprise_id}@geofence@.yaml:58]: What is uploadGeofence?

Comment (BindyaBhaskar) [reply spec/paths/v0@enterprise@{enterprise_id}@geofence@.yaml:58]: Creating a geofence is what I meant.

**Review by abdulMuneerM — APPROVED** [review]

**Pull request merged**